### PR TITLE
Fix warning sign compare

### PR DESCRIPTION
**Pull request by Bogumil-Sapinski-Mobica**

### CODE REVIEW

File: lib/include/CL/SDK/Image.hpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace cl {
 namespace sdk {
     struct Image
     {
-        int width = 0, height = 0, pixel_size = 1;
+        size_t width = 0, height = 0, pixel_size = 1;
```
Comment (alycm) [inline lib/include/CL/SDK/Image.hpp:13]: Given there are changes below to `reinterpret_cast` these back to `int` in one location below is this the correct change?  Should the other side of whatever was triggering the warning be changed?

**Review by MathiasMagnus — CHANGES_REQUESTED** [review]
Mostly good, but the `sdk::Image` changes I'd need convincing about.

**Comment by Bogumil-Sapinski-Mobica** [comment]
Hi @MathiasMagnus,
I have changed solution for warnings caused by comparing signed and unsigned ints. Please verify if this is ok for you.

**Review by MathiasMagnus — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as discussed in the September 12th teleconference.  The CI failures are unrelated to these changes.

**Pull request merged**